Improve the same user check and its documentation (#69844)
elastic/elasticsearch: docs/reference/search/scroll-api.asciidoc

The search response returns a scroll ID in the `_scroll_id` response body
parameter. You can then use the scroll ID with the scroll API to retrieve the
next batch of results for the request. If the {es} {security-features} are enabled,
the access to the results of a specific scroll ID is restricted to

> <<same-user-check,the user that submitted the search in the first place>>.

Reviewer: ```suggestion
<<same-user-check,the user or API Key that submitted the search>>.
```

> <<can-access-resources-check,the user or API Key that submitted the search>>.

Reviewer 2: ```suggestion
<<can-access-resources-check,the user or API key that submitted the search>>.
```
